src/test/java/com/puppycrawl/tools/checkstyle/DefaultConfigurationTest.java

```diff
@@ -19,15 +19,43 @@
 
 package com.puppycrawl.tools.checkstyle;
 
+import static org.junit.Assert.assertArrayEquals;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
 
+import java.util.Map;
+import java.util.TreeMap;
+
 import org.junit.Test;
 
 import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
 
 public class DefaultConfigurationTest {
 
+    @Test
+    public void testGetAttributeNames() {
+        final DefaultConfiguration config = new DefaultConfiguration("MyConfig");
+        config.addAttribute("attribute", "value");
+        final String[] actual = config.getAttributeNames();
+        final String[] expected = {"attribute"};
+        assertArrayEquals("Invalid attribute names", expected, actual);
+    }
+
+    @Test
+    public void testAttribute() throws CheckstyleException {
```

Comment:
This is named `testAttribute` and tests `getAttribute`.
`testGetAttributeNames` tests `getAttributeNames`.
`testGetName` tests `getName`.

Why does this one break consistency of method names?
Either this should be `testGetAttribute` or others should be named `testName`, etc...

Reply:
Because it tests both `addAttribute` and `getAttribute`. The name `testAddAttributeAndGetAttribute` looks worse for me then simple `testAttribute`.

Reply:
testAddAttributeAndGetAttribute looks better for me

Reply:
done